### ISSUE TITLE
Zastosuj Allegro API

### ISSUE THREAD

**Issue by Lebioda** [MK-PL/AllegroRSS#1]
Zamiast scrapować allegro można użyć Allegro API i zastosować metodę doGetItemsList. Szybciej, prościej i pewniej :)

**Comment by MK-PL** [comment]
Dziękuję za sugestię :)

Pytanie tylko, czy Allegro API umożliwia filtrowanie wyników wyszukiwania po opcjach np. koloru czy rozmiaru? Z tego, co testowałem, to metoda doGetItemsList tego nie ma - są tylko domyślne opcje typu stan, rodzaj oferty, lokalizacja :/

**Comment by Lebioda** [comment]
Tak, dostępne są wszystkie filtry. Mam np. skrypt, który powiadamia o nowych ofertach na używane, uszkodzone samochody, które mają kierownicę z lewej strony. Dostępne filtry pobiera się tą samą metodą.

**Comment by MK-PL** [comment]
> Dostępne filtry pobiera się tą samą metodą.

Co trzeba podać na wejściu, żeby te filtry (np. o rozmiarze, czy kolorze) były widoczne w danych zwracanych?

Do wyszukiwania chcę wykorzystać tylko adres strony, w którym są zamieszczone zmienne GET od filtrów wyszukiwania i na podstawie tego adresu chcę pobrać aukcje w Allegro API (żeby wynik zgadzał się z tym, co jest w przeglądarce). Problem jest też w tym, że zmienne GET używane w adresie strony różnią się od tych używanych w WebApi np. w adresie jest string dla frazy wyszukiwania, a w WebAPI jest 'filterId' => 'search', ale w przypadku dodatkowych filtrów powinny być wtedy dostępne etykiety typu [filterName] => Cena, po których będzie można dopasować filtr (jeśli dobrze to zrozumiałem?).

**Comment by Lebioda** [comment]
Zwracane są filtry adekwatne do wybranej kategorii. Jeśli nie ma w zapytaniu nr kategorii to zwracane są tylko globalne filtry. Czyli jeśli zapytamy o kategorię Samochody Osobowe (nr 4029) to zwróci filtry dla tej kategorii.

Jeśli ma działać tylko na podstawie linku to będzie potrzebne zrobienie jakieś tabelki dopasowującej, bo niestety nazwy nie są takie same, a większość filtrów ma tylko identyfikator liczbowy.

**Comment by Lebioda** [comment]
Zmiany w Allegro API:
https://developer.allegro.pl/news/2018-07-03-Listing_ofert/#1

Może warto zainteresować się tym tematem, aby skrypt był bardziej niezawodny :-)

![screenshot_144](https://user-images.githubusercontent.com/5007985/42413738-f5af624c-8226-11e8-9ae3-9bcdb652b14b.jpg)

**Comment by Lebioda** [comment]
W grudniu Allegro dodało nową opcje w REST API:

**Client_credentials flow dla Allegro OAuth2**
W ramach Allegro OAuth2 udostępniliśmy ścieżkę, dzięki której możesz autoryzować się bez zgody użytkownika na działanie. Wybierając ten sposób uzyskasz dostęp do zasobów, które nie wymagają kontekstu użytkownika (pobieranie listy kategorii, przeglądanie listingu itp.)

https://developer.allegro.pl/auth/#clientCredentialsFlow

Teraz można bez problemu zrobić Generator RSS bazując na API, do dyspozycji są wszystkie filtry widoczne normalnie przez stronę.
